tests/unit/test_repository.py

```diff
@@ -454,7 +454,7 @@ def test_get_pipelines_forwards_to_metadata_store(clean_repo, mocker):
         metadata_store=new_metadata_store,
         artifact_store=active_stack.artifact_store,
     )
-    clean_repo.register_stack(new_stack)
+    clean_repo.register_stack(new_stack, reset_association=True)
```

Comment:
I definitely feel like this feature will need to have some unit tests for the reset, for the failure in case a previously associated artifact store is reused, etc ...

Reply:
You are absolutely right. I am running into some issues regarding the tests at the moment. As soon as I resolve them, I will write some specific tests for this interaction as well!